lib/plugins/aws/deployList/index.js

```diff
@@ -61,6 +66,77 @@ class AwsDeployList {
         return BbPromise.resolve();
       });
   }
+
+  // list all functions and their versions
+  listFunctions() {
+    return BbPromise.resolve().bind(this)
+      .then(this.getFunctions)
+      .then(this.getFunctionVersions)
+      .then(this.displayFunctions);
+  }
+
+  getFunctions() {
+    const params = {
+      MaxItems: 200,
+    };
+
+    return this.provider.request('Lambda',
+      'listFunctions',
+      params,
+      this.options.stage,
+      this.options.region)
+      .then((result) => {
+        const allFuncs = result.Functions;
+
+        const serviceName = `${this.serverless.service.service}-${this.options.stage}`;
+        const regex = new RegExp(serviceName);
+        const serviceFuncs = allFuncs.filter((func) => func.FunctionName.match(regex));
+
+        return BbPromise.resolve(serviceFuncs);
+      });
+  }
+
+  getFunctionVersions(funcs) {
+    const requestPromises = [];
+
+    funcs.forEach((func) => {
+      const params = {
+        FunctionName: func.FunctionName,
+        MaxItems: 5,
```

Comment:
We only keep 5 versions of the artifacts in the S3 bucket so this aligns with that.

We could display more though...